### DISCUSSION TITLE
How to Run Powershell Script from Extension

### DISCUSSION THREAD

**Discussion by Denisjc** [VsixCommunity/Community.VisualStudio.Toolkit#481] [Q&A] [answered]
How would you run a script from the Extension.

As I am battling to find scripts that I included in the path for the extension and so cant call then via a path name.

Any ideas?

**Comment by Denisjc** [comment]
I seem to have located how to call a script
This is my code for the base directory
` var basePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);`

I have set the scripts to be copy always and they appear under this location.
Will this be the same for runtime when the Extension is installed?

**Reply by reduckted** [reply, marked as answer]
Yep, that's correct. Here's an example of an extension that does the same sort of thing. It includes "template" files in the VSIX and loads them at runtime.

https://github.com/madskristensen/AddAnyFile/blob/8a8443fbf0f5953cf83849a489350b0e42ca2e86/src/Templatemap.cs#L30
